blockprod/src/detail/mod.rs

```diff
@@ -254,12 +260,47 @@ impl BlockProduction {
             self.job_manager.make_job_stopper_function();
         let _job_stopper_destructor = OnceDestructor::new(move || job_stopper_function(job_key));
 
+        // Unlike Proof of Work, which can vary any header field when
+        // searching for a valid block, Proof of Stake can only vary
+        // the header timestamp. Its search space starts at the
+        // previous block's timestamp + 1 second, and ends at the
+        // current timestamp + some distance in time defined by the
+        // blockchain.
+        //
+        // This variable keeps track of the last timestamp that was
+        // attempted, and during Proof of Stake, will prevent
+        // searching over the same search space.
+        let last_epoch_used = {
```

Comment:
I think we shouldn't call this epoch... it's confusing because we already have that word used elsewhere.

Reply:
I just noticed that everything was renamed to epoch... why?

Reply:
I'm used to calling any integer that represents seconds since Epoch as the unit "epoch". I've renamed "epoch" to "timestamp_seconds".